src/brokers/commerzbank.js

```diff
@@ -88,81 +96,100 @@ const findCompanyBuy = textArr => {
+  let type,
+    date,
+    wkn,
+    isin,
+    company,
+    shares,
+    price,
+    amount,
+    fxRate,
+    foreignCurrency;
+  let fee = 0;
+  let tax = 0;
   let activity;
   if (isBuy(textArr)) {
-    const foundAmount = Big(findAmountBuy(textArr));
-    const foundShares = findSharesBuy(textArr);
-    activity = {
-      broker: 'commerzbank',
-      type: 'Buy',
-      date: findDateBuy(textArr),
-      wkn: findWknBuy(textArr),
-      company: findCompanyBuy(textArr),
-      shares: +foundShares,
-      price: +foundAmount.div(foundShares),
-      amount: +foundAmount,
-      fee: 0,
-      tax: 0,
-    };
+    type = 'Buy';
+    date = findDateBuy(textArr);
+    wkn = findWknBuy(textArr);
+    company = findCompanyBuy(textArr);
+    shares = +findSharesBuy(textArr);
+    amount = findAmountBuy(textArr);
+    price = findPriceBuy(textArr);
+    fee = findFeeBuy(textArr, amount);
   } else if (isDividend(textArr)) {
-    const foundAmount = Big(findAmountDividend(textArr));
-    const foundShares = findSharesDividend(textArr);
-    activity = {
-      broker: 'commerzbank',
-      type: 'Dividend',
-      date: findDateDividend(textArr),
-      wkn: findWknDividend(textArr),
-      isin: findIsinDividend(textArr),
-      company: findCompanyDividend(textArr),
-      shares: +foundShares,
-      price: +foundAmount.div(foundShares),
-      amount: +foundAmount,
-      fee: 0,
-      tax: +findTaxDividend(textArr),
-    };
-  } else if (isForeignDividend(textArr)) {
-    const foundAmount = Big(findAmountForeignDividend(textArr));
-    const foundShares = findSharesForeignDividend(textArr);
-    activity = {
-      broker: 'commerzbank',
-      type: 'Dividend',
-      date: findDateForeignDividend(textArr),
-      wkn: findWknForeignDividend(textArr),
-      isin: findIsinForeignDividend(textArr),
-      company: findCompanyForeignDividend(textArr),
-      shares: +findSharesForeignDividend(textArr),
-      price: +foundAmount.div(foundShares),
-      amount: +foundAmount,
-      fee: 0,
-      tax: 0,
-    };
+    const foreignCurrencyIndex = textArr.indexOf('Devisenkurs:');
+    const foreignDividend = foreignCurrencyIndex >= 0;
+    type = 'Dividend';
+    date = findDateDividend(textArr, foreignDividend);
+    wkn = findWknDividend(textArr);
+    isin = findIsinDividend(textArr, foreignDividend);
+    company = findCompanyDividend(textArr, foreignDividend);
+    shares = findSharesDividend(textArr);
+    amount = findAmountDividend(textArr, foreignDividend);
+    price = +Big(amount).div(shares);
+    tax = findTaxDividend(textArr);
+    if (foreignCurrencyIndex >= 0) {
+      foreignCurrency = textArr[foreignCurrencyIndex + 1].split('/')[1];
+      fxRate = parseGermanNum(textArr[foreignCurrencyIndex + 2]);
+    }
+  }
+  // sadly, no exact times can be extracted as they are not given in any of the
+  // files
+  const [parsedDate, parsedDateTime] = createActivityDateTime(
+    date,
+    undefined,
+    'dd.MM.yyyy',
```

Comment:
The date format is the default one. time is undefined, so we can remove both parameters.
```suggestion
    createActivityDateTime(date, undefined)
```

(same below for transaction reports)


This is more a minor think. I'm also fine with the current version.

Reply:
Done, ty